src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -1095,6 +1090,122 @@ public void Deconstruct(out int a, out int b)
         }
 
         [Fact]
+        [WorkItem(18629, "https://github.com/dotnet/roslyn/issues/18629")]
+        public void ValueTupleNotRequiredIfReturnIsNotUsed()
```

Comment:
>ValueTupleNotRequiredIfReturnIsNotUsed [](start = 20, length = 38)

Should probably have tests to verify that obsolete diagnostics is suppressed even when the type is present.